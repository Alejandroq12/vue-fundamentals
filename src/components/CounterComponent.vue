<template>
    <h2>{{ customTitle }} </h2>
    <p> {{ counter }} <sup>2</sup> = {{ squareCounter }} </p>

    <div>
        <button @click="increase">+1</button>
        <button @click="decrease">-1</button>
    </div>

</template>

<script>
export default {
    props: {
        title: String,
        start: {
            type: Number,
            default: 100,
            // required: true
            validator( value ) {
                return value >= 0
            }
        }
    },
    // name: 'Patito'
    data() {
        return {
            counter: this.start
        }
    },
    methods: {
        getSquareValue() {
            return this.counter * this.counter
        },
        increase() {
            this.counter++
        },
        decrease() {
            this.counter--
        },
    },
    computed: {
        squareCounter() {
            return this.counter * this.counter
        },
        customTitle() {
            return this.title || 'Counter'
        }
    }
}
</script>

<style>

button{
    background-color: #64B687;
    border-radius: 5px;
    border: 1px solid white;
    color: white;
    cursor: pointer;
    margin: 0 5px;
    padding: 5px 15px;
    transition: 0.3s ease-in-out;
}

button:hover {
    background-color: #5aa67b;
    transition: 0.3s ease-in-out;
}

</style>