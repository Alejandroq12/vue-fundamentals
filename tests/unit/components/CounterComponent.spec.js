import { shallowMount } from '@vue/test-utils';
import CounterComponent from '../../../src/components/CounterComponent.vue';

describe('Counter Component', () => {
  // test('debe de hacer match con el snapshot', () => {
  //     const wrapper = shallowMount(CounterComponent)
  //     expect(wrapper.html()).toMatchSnapshot()
  // })
  test('h2 debe de tener el valor por defecto.', () => {
    const wrapper = shallowMount(CounterComponent);
    expect(wrapper.find('h2').exists()).toBeTruthy();
    const h2Value = wrapper.find('h2').text();
    expect(h2Value).toBe('Counter');
  });

  test('Default value must be 100', () => {
    const wrapper = shallowMount(CounterComponent);
    const value = wrapper.find('[data-testid="counter"]').text();
    expect(value).toBe('100');
  });

  test('It must be increased and decrease the value of the counter', async () => {
    const wrapper = shallowMount( CounterComponent );
    const increaseBtn = wrapper.find('button');
    await increaseBtn.trigger('click');
    let value = wrapper.find('[data-testid="counter"]').text()
    expect( value ).toBe('101');

    const decreaseBtn = wrapper.findAll('button')[1];
    await decreaseBtn.trigger('click');
    await decreaseBtn.trigger('click');
    value = wrapper.find('[data-testid="counter"]').text()
    expect( value ).toBe('99');
  })
});
