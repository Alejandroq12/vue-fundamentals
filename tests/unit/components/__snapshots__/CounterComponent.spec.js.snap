// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component debe de hacer match con el snapshot 1`] = `
<h2>Counter</h2>
<p>100 <sup>2</sup> = 10000</p>
<p data-testid="counter">100</p>
<div class="button-container"><button>+1</button><button>-1</button></div>
`;
